src/OrchardCore.Modules/OrchardCore.Localization/Views/CulturePickerContainer.cshtml

```diff
@@ -0,0 +1,27 @@
+@using OrchardCore.Localization
+@using OrchardCore.Environment.Shell
+@inject ShellSettings ShellSettings
+
+@{
+    var selectedCultureName = Model.CurrentCulture.Name;
+}
+<select id="culture-picker" class="form-select me-3" data-selected-culture="@selectedCultureName">
+  @foreach (var culture in Model.SupportedCultures)
+  {
+    <option selected="@(culture.Name == selectedCultureName ? "selected" : null)" value="@culture.Name">@culture.DisplayName</option>
+  }
+</select>
+
+<script at="Foot" depends-on="jquery">
+    $(function (){
+        $('#culture-picker').change(function(){
+            var culture = $(this).val();
+            var selectedCulture = $(this).data('selected-culture');
+            if (culture != selectedCulture) {
+                var cookieValue = `@LocalizationCookieName.AdminSite=c=${culture}|uic=${culture};path=/@ShellSettings.Name.ToLower()`;
+                document.cookie = cookieValue;
+                window.location = window.location.href.split('?')[0];
```

Comment:
Some suggestions mainly based on what we are used to do, and as I remember ;)

A more specific picker id `#culture-picker` => `#oc-admin-culture-picker`.

Good to retrieve the `selected-culture` from a data attribute, better do the same for all other data without using the razor `@` operator, and that you could init once for the static ones just before the jQuery `change()` code line.

About the cookie path would be better to use the `RequestUrlPrefix` in some place we use e.g.

    _tenantPrefix = "/" + shellSettings.RequestUrlPrefix;
    options.Cookie.Path = _tenantPrefix;

But this not totally right as it is not well url escaped, better to just use the current request `PathBase`, not its `.Value` which is the original non escaped value, but its `.ToString()` which is implicitly called when resolving a string.

    path = pathBase.HasValue() ? pathBase : "/";

Note: Looks like in some place aspnetcore adds this path to the cookie as I saw e.g. for the antiforgery cookie for which we don't specify explicitly any path. Maybe it would have worked by just setting the path to `/` but then also differentiate the cookie name per tenant.

Anyway we also have to differentiate the cookie name per tenant, in some place we do e.g.

    CurrentTenant = shellSettings.Name;
    var cookieName = $"{CurrentTenant}-adminPreferences";

But this not totally right as the tenant name may contain chars that are not allowed for a cookie name, as we experienced for the antiforgery cookie name, that's why we now use the shell `VersionId` instead.

    var cookieName = "__orchantiforgery_" + settings.VersionId;

Finally for now, not sure it is good to strip the query string part when redirecting, here maybe better to retrieve the current `FullRequestPath` which is a property of our `RazorPage`.

Reply:
Thanks @jtkech for your information, I agree with what you mentioned expect using different cookie name per tenant, the path is a right option IMHO. I already saw such thing in light/dark theme if I remember

Reply:
Woulld need to be tried

About the darkmode we use the following cookie with different names per tenant

            CurrentTenant = shellSettings.Name;
            var cookieName = $"{CurrentTenant}-adminPreferences";

Other examples

                options.Cookie.Name = "orchauth_" + HttpUtility.UrlEncode(_tenantName);

                var cookieName = "__orchantiforgery_" + settings.VersionId;

Here we don't set the cookie path but it seems that aspnet set it for us e.g. `/` or `/tenant1`. So maybe when in the context of the default tenant it could take into account the cookie of the tenant1 if `/tenant1` is just considered as a subfolder (would need to test it) and if we don't differentiate the names.

About using the above ways we experencied some issues for not allowed char in cookie names with some special chars in the tenant name, that' why I recommend this pattern.

                var cookieName = "anyAllowedChars-" + settings.VersionId;

Note: prefixing with a double undescores is for additional security policies that we don't need here.